src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -92,12 +189,332 @@ struct FixedKVInternalNode
+    tracker->child = child.weak_from_this();
+  }
+
+  void new_child(Transaction &t, CachedExtent &child, uint64_t pos) {
+    LOG_PREFIX(FixedKVInternalNode::new_child);
+    auto &tracker = this->child_trackers[pos];
+    assert(tracker != nullptr);
+    ceph_assert(child.is_mutation_pending() && this->is_pending());
+    t.trackers_to_rm.push_back(tracker);
+    tracker = new child_tracker_t(&child);
+    t.new_pending_trackers.push_back(tracker);
+    SUBTRACET(seastore_fixedkv_tree,
+      "new tracker: {}, this: {}, child: {}",
+      t, (void*)tracker, *this, child);
+  }
+
+  void add_child_trans_view(FixedKVNode<NODE_KEY> &child, uint64_t pos) {
+    ceph_assert(pos < this->get_size());
+    this->child_trans_views.new_trans_view(child, pos);
+
+    ceph_assert(!this->is_pending() && child.is_mutation_pending());
+    ceph_assert(!child.parent_tracker);
+    ceph_assert(this->child_trackers[pos]);
+    child.parent_tracker = std::make_unique<parent_tracker_t>(this, pos);
+  }
+
+  // this method should only be invoked to rewrite extents
+  void copy_child_trackers_out(
+    Transaction &t,
+    FixedKVInternalNode &new_node) {
+    LOG_PREFIX(FixedKVInternalNode::copy_child_trackers_out);
+    SUBTRACET(seastore_fixedkv_tree,
+      "coping {} trackers from {} to {}",
+      t, this->get_size(), *this, new_node);
+
+    ceph_assert(this->get_type() == new_node.get_type());
+    auto data = this->child_trackers.data();
+    auto n_data = new_node.child_trackers.data();
+#ifndef NDEBUG
+    for (int i = 0; i < this->get_size(); i++) {
+      assert(this->child_trackers[i]);
+    }
+#endif
+    std::memmove(n_data, data, this->get_size() * sizeof(child_tracker_t*));
+    if (!this->is_pending()) {
+      auto children = this->child_trans_views
+	.template remove_trans_view<base_t>(t);
+      for (auto [child, pos] : children) {
+	auto &tracker = new_node.child_trackers[pos];
+	tracker = new child_tracker_t(child);
+	((base_t*)child)->parent_tracker.reset();
+	t.new_pending_trackers.push_back(tracker);
+	t.trackers_to_rm.push_back(this->child_trackers[pos]);
+	SUBTRACET(seastore_fixedkv_tree,
+	  "new tracker: {}, this: {}, child: {}",
+	  t, (void*)new_node.child_trackers[pos], new_node, *child);
+      }
+    }
+  }
+
+  void split_child_trackers(
+    Transaction &t,
+    node_type_t &left,
+    node_type_t &right)
+  {
+    ceph_assert(left.get_size() > 0);
+    ceph_assert(right.get_size() > 0);
+    LOG_PREFIX(FixedKVNode::split_child_trackers);
+    size_t pivot = this->get_size() / 2;
+    child_tracker_t** l_data = left.child_trackers.data();
+    child_tracker_t** r_data = right.child_trackers.data();
+    child_tracker_t** data = this->child_trackers.data();
+    size_t l_size = pivot;
+    size_t r_size = this->get_size() - pivot;
+
+    std::memmove(l_data, data, sizeof(child_tracker_t*) * l_size);
+    std::memmove(r_data, data + pivot, sizeof(child_tracker_t*) * r_size);
```

Comment:
Um, this memmove is actually copying data, not moving them.

Reply:
std::memcpy would seem to be the better option here since the ranges do not overlap.  std::memmove will work, but will be less performant due to needing to tolerate overlapping source and destination ranges.